src/importexport/midi/internal/midiexport/exportmidi.cpp

```diff
@@ -370,6 +371,31 @@ bool ExportMidi::write(QIODevice* device, bool midiExpandRepeats, bool exportRPN
                 }
             }
         }
+
+        // Export lyrics
+        SegmentType st = SegmentType::ChordRest;
+        for (Segment* seg = m_score->firstMeasure()->first(st); seg; seg = seg->next1(st)) {
+            for (track_idx_t i = part->startTrack(); i < part->endTrack(); ++i) {
+                ChordRest* cr = toChordRest(seg->element(i));
+                if (cr) {
+                    for (const auto& lyric : cr->lyrics()) {
+                        ByteArray lyricText = lyric->plainText().toUtf8();
+                        size_t len = lyricText.size() + 1;
+                        unsigned char* data = new unsigned char[len];
```

Comment:
Looks like memory leak here. According to the fact that this code used in the loop there is huge memory leak possible =)
Isn't there a method on mu::String which returns Char*?

Reply:
Note that this method is full of similar memory leaks, therefore I ignored it while reviewing.

> Isn't there a method in mu::String which returns char*?

No, because mu::String uses utf16 so if you want char* you need to convert.



> Or just `delete[] data;` at the end of that loop

Hell no 😄 because we're storing a pointer to it in the resulting MidiEvent.

I think the conclusion is that we should refactor to avoid the need for char*, and store it differently in MidiEvent.

Reply:
```size_t len = lyricText.size();  // Not sure + 1 is necessary. If so it should be data[len] = '\0', otherwise it's undefined```

However why not to use easier approach without unnecessary memory allocation?

```cpp
std::string lyricText = lyric->plainText().toStdString();
size_t len = lyricText.size();
ev.setEData(lyricText.c_str());
ev.setLen(static_cast<int>(len));
```

Reply:
> Hell no 😄 because we're storing a pointer to it in the resulting MidiEvent.

Good point

Reply:
> ```c++
> std::string lyricText = lyric->plainText().toStdString();
> size_t len = lyricText.size();
> ev.setEData(lyricText.c_str());

`c_str()` is not an `unsigned char *` like `setEData()` expects

Reply:
But even with that it just doesn't work, on importing such a MIDI the lyrics are completly broken

Reply:
> `size_t len = lyricText.size(); // Not sure + 1 is necessary. If so it should be data[len] = '\0', otherwise it's undefined`
> 
> However why not to use easier approach without unnecessary memory allocation?
> 
> ```c++
> std::string lyricText = lyric->plainText().toStdString();
> size_t len = lyricText.size();
> ev.setEData(lyricText.c_str());
> ev.setLen(static_cast<int>(len));
> ```

That's basically equivalent to using `delete`: as soon as `lyricText` goes out of scope, the underlying c string is deleted but the MidiEvent still contains a pointer to it. That's probably why the result is "completely broken" on Windows and would cause Address Sanitizer complaints on macOS/Linux.